sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala

```diff
@@ -215,7 +215,7 @@ class StreamingQueryManager private[sql] (sparkSession: SparkSession) extends Lo
       recoverFromCheckpointLocation: Boolean,
       trigger: Trigger,
       triggerClock: Clock): StreamingQueryWrapper = {
-    var deleteCheckpointOnStop = false
+    var deleteCheckpointOnStop = sparkSession.sessionState.conf.isDeleteCheckpoint.getOrElse(false)
```

Comment:
I checked the code, I think the origin design do not want to delete the checkpoint directory unless you specify one.
If you want control delete behavior, not specify the checkpoint directory. Temp directory will be delete after app stop.